com.unity.multiplayer.mlapi/Prototyping/NetworkTransform.cs

```diff
@@ -12,282 +14,180 @@ namespace MLAPI.Prototyping
+        public NetworkChannel Channel = NetworkChannel.NetworkVariable;
 
         /// <summary>
-        /// The channel to send the data on
+        /// Sets whether this transform should sync in local space or in world space.
+        /// This is important to set since reparenting this transform could have issues,
+        /// if using world position (depending on who gets synced first: the parent or the child)
+        /// Having a child always at position 0,0,0 for example will have less possibilities of desync than when using world positions
         /// </summary>
-        [SerializeField, Tooltip("The channel to send the data on.")]
-        public NetworkChannel Channel = NetworkChannel.NetworkVariable;
+        [Tooltip("Sets whether this transform should sync in local space or in world space")]
+        public bool InLocalSpace = false;
 
         /// <summary>
-        /// Sets whether this transform should sync local or world properties. This is important to set since reparenting this transform
-        /// could have issues if using world position (depending on who gets synced first: the parent or the child)
-        /// Having a child always at position 0,0,0 for example will have less possibilities of desync than when using world positions
+        /// The base amount of sends per seconds to use when range is disabled
         /// </summary>
-        [SerializeField, Tooltip("Sets whether this transform should sync local or world properties. This should be set if reparenting.")]
-        private NetworkVariableBool m_UseLocal = new NetworkVariableBool();
+        [SerializeField, Range(0, 120), Tooltip("The base amount of sends per seconds to use when range is disabled")]
+        public float FixedSendsPerSecond = 30f;
 
-        public bool UseLocal
-        {
-            get => m_UseLocal.Value;
-            set => m_UseLocal.Value = value;
-        }
+        private readonly NetworkVariable<NetworkState> m_NetworkState = new NetworkVariable<NetworkState>(new NetworkState());
 
         /// <summary>
-        /// Updates the NetworkTransform's authority model at runtime.
+        /// Does this instance (client or server) has authority to update transform?
         /// </summary>
-        /// <param name="newAuthority"></param>
-        public void SetAuthority(Authority newAuthority)
-        {
-            TransformAuthority = newAuthority;
-            UpdateVarPermissions();
-            // todo this should be synced with the other side. let's wait for a more final solution before adding more code here
-        }
+        public bool CanUpdateTransform =>
+            Authority == NetworkAuthority.Client && IsClient && IsOwner ||
+            Authority == NetworkAuthority.Server && IsServer ||
+            Authority == NetworkAuthority.Shared;
 
-        private void UpdateOneVarPermission<T>(NetworkVariable<T> varToUpdate)
+        private bool IsNetworkStateDirty
```

Comment:
note: implementing threshold values would hopefully be easier with this property's body. we could simply ignore too small changes and count them as "not dirty"